tests/e2e/index_auto_creation/index_auto_creation.py

```diff
@@ -0,0 +1,318 @@
+
+    execute_and_fetch_all(cursor, f"MATCH (n) DETACH DELETE n")
+
+
+def test_auto_create_single_edge_type_index_with_multiple_entries(cursor):
+    label_from = "LABEL_FROM"
+    label_to = "LABEL_TO"
+    edge_type = "SOMEEDGETYPE"
+    execute_and_fetch_all(cursor, f"CREATE (n:{label_from})")
+    execute_and_fetch_all(cursor, f"CREATE (n:{label_to})")
+    for _ in range(100):
+        execute_and_fetch_all(cursor, f"MATCH (n:{label_from}), (m:{label_to}) CREATE (n)-[:{edge_type}]->(m)")
+
+    index_stats = get_index_stats(cursor)
+    assert index_exists(index_stats, label_from)
+    assert index_exists(index_stats, label_to)
+    assert index_exists(index_stats, edge_type)
+    assert number_of_index_structures_are(index_stats, 3)  # 2 label + 1 edge-type
+
+    assert index_count_is(index_stats, edge_type, 100)
+
+    execute_and_fetch_all(cursor, f"DROP INDEX ON :{label_from}")
+    execute_and_fetch_all(cursor, f"DROP INDEX ON :{label_to}")
+    execute_and_fetch_all(cursor, f"DROP EDGE INDEX ON :{edge_type}")
+    index_stats = get_index_stats(cursor)
+    assert number_of_index_structures_are(index_stats, 0)
+
+    execute_and_fetch_all(cursor, f"MATCH (n) DETACH DELETE n")
+
+
+def test_auto_create_multiple_edge_type_index_with_multiple_entries(cursor):
+    label_from = "LABEL_FROM"
+    label_to = "LABEL_TO"
+    edge_type1 = "SOMEEDGETYPE1"
+    edge_type2 = "SOMEEDGETYPE2"
+    edge_type3 = "SOMEEDGETYPE3"
+
+    execute_and_fetch_all(cursor, f"CREATE (n:{label_from})")
+    execute_and_fetch_all(cursor, f"CREATE (n:{label_to})")
+    for _ in range(120):
+        execute_and_fetch_all(cursor, f"MATCH (n:{label_from}), (m:{label_to}) CREATE (n)-[:{edge_type1}]->(m)")
+    for _ in range(100):
+        execute_and_fetch_all(cursor, f"MATCH (n:{label_from}), (m:{label_to}) CREATE (n)-[:{edge_type2}]->(m)")
+    for _ in range(80):
+        execute_and_fetch_all(cursor, f"MATCH (n:{label_from}), (m:{label_to}) CREATE (n)-[:{edge_type3}]->(m)")
+
+    index_stats = get_index_stats(cursor)
+
+    assert number_of_index_structures_are(index_stats, 5)  # 2 label + 3 edge-type
+    assert index_exists(index_stats, label_from)
+    assert index_exists(index_stats, label_to)
+    assert index_exists(index_stats, edge_type1)
+    assert index_exists(index_stats, edge_type2)
+    assert index_exists(index_stats, edge_type3)
+    assert index_count_is(index_stats, edge_type1, 120)
+    assert index_count_is(index_stats, edge_type2, 100)
+    assert index_count_is(index_stats, edge_type3, 80)
+
+    execute_and_fetch_all(cursor, f"DROP INDEX ON :{label_from}")
+    index_stats = get_index_stats(cursor)
+    assert number_of_index_structures_are(index_stats, 4)
+
+    execute_and_fetch_all(cursor, f"DROP INDEX ON :{label_to}")
+    index_stats = get_index_stats(cursor)
+    assert number_of_index_structures_are(index_stats, 3)
+
+    execute_and_fetch_all(cursor, f"DROP EDGE INDEX ON :{edge_type1}")
```

Comment:
Do you intentionally not use loop to drop edge indices?

Reply:
IMO this is more readable for tests. I can change it if you insist.